### PR TITLE
etcdserver/etcdhttp/httptypes: handle http write errors

### PR DESCRIPTION
**Pull request by gyuho**
From `TODO(xiangli): handle http write errors`,
this logs http write errors and logs with
`github.com/coreos/pkg/capnslog`.

Thanks!

### CODE REVIEW

File: etcdserver/etcdhttp/httptypes/errors.go [hunk 1]
```diff
@@ -43,7 +43,10 @@ func (e HTTPError) WriteTo(w http.ResponseWriter) {
 	if err != nil {
 		plog.Panicf("marshal HTTPError should never fail (%v)", err)
 	}
-	w.Write(b)
+	if _, err := w.Write(b); err != nil {
+		plog.Errorf("http write errors with HTTPError (%v)", err)
```
Comment (xiang90) [inline etcdserver/etcdhttp/httptypes/errors.go]: After thinking for a while, I think we probably make this function return an error.

I want to log like this `error writing HTTPError ("ABC") to 10.0.0.1:12345`. However, only the caller know the remote Addr.

File: etcdserver/etcdhttp/httptypes/errors.go [hunk 2]
```diff
@@ -43,7 +43,10 @@ func (e HTTPError) WriteTo(w http.ResponseWriter) {
...
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```
Comment (xiang90) [inline etcdserver/etcdhttp/httptypes/errors.go]: I think the write error can only be caused by a broken connection or write timeout. Not sure if it is still reasonable to retry writing to the same client again. I would suggest just return here.

File: etcdserver/etcdhttp/httptypes/errors.go (hunk 1 above)
Comment (xiang90) [inline etcdserver/etcdhttp/httptypes/errors.go]: Also we probably should make this a debug level. A client side error can cause this too. This is not a server side error in many cases.

**Comment by gyuho** [comment]
@xiang90 Thanks for the review!
Ok, I see your point. I will push the changes as you suggested tonight.

**Comment by gyuho** [comment]
Closing this in favor of https://github.com/coreos/etcd/pull/3579.

Thanks!

**Pull request closed without merging**